src/uct/ib/ud/base/ud_ep.c

```diff
@@ -792,7 +792,10 @@ ucs_status_t uct_ud_ep_flush(uct_ep_h ep_h, unsigned flags,
         return UCS_OK;
     }
 
-    UCT_UD_IFACE_CHECK_ASYNC_PENDING(iface, UCS_ERR_NO_RESOURCE);
+    if (ucs_unlikely(uct_ud_iface_has_pending_async_ev(iface))) {
+        uct_ud_leave(iface);
+        return UCS_ERR_NO_RESOURCE;
```

Comment:
maybe use 'goto' here, and add label before calling uct_ud_leave() in the end of the function?